### PR TITLE
To Do List: Code Review

### PR DESCRIPTION
**Pull request by Recillah-Khamala**
### **To Do List: Code Review**

In this pull request, I have copied my code from the "To Do list" project and committed it as one big chunk of code in the feature branch for my partner to review it.

### CODE REVIEW

**Comment by samlorlah** [comment]
Hi @Rn486 ✋,

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there!




### Highlights 💪

- Descriptive pull request ✔️
- No linter errors ✔️
- Good presentation demo ✔️
- JavaScript best practices ✔️
- HTML and CSS best practices ✔️
- Good use of DRY, KISS, and YAGNI rules ✔️

### Required Changes ♻️
Check the comments under the review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: src/crud.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  const input = add.value;
+  e.preventDefault();
+  add.value = '';
+
+  if (!input) return;
+  const obj = {
+    description: input,
+    completed: false,
+    index: store.length,
```
Comment (samlorlah) [inline src/crud.js:37]: - [x] Kindly update your index to start from 1 according to the instruction. Update your code at this point to have your task index as requested

File: src/crud.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+window.onload = () => {
+  if (localStorage.getItem('store')) {
+    const storage = JSON.parse(localStorage.getItem('store'));
+    storage.forEach((val) => {
+      val.completed = false;
+    });
+    localStorage.setItem('store', JSON.stringify(storage));
+  }
+};
```
Comment (samlorlah) [inline src/crud.js:100]: - [ ] When the window refresh button is pressed, the local storage state should not be altered. Kindly reverse the code.

File: src/crud.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    const index = (e.target.id);
+    removeTask(index);
+  }
+});
+
+export const update = (e) => {
+  const clicked = e.target.closest('.toDo');
+  if (!clicked) return;
+  clicked.addEventListener('keyup', () => {
+    // const item = parseInt(clicked.getAttribute('data-desc', 10));
```
Comment (samlorlah) [inline src/crud.js:66]: - [x] According to YAGNI rule, this should be removed please

File: src/crud.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            <div class="icon">
+            <i class="fa-solid fa-trash-can" id="${element.index}"></i>
+            </div>`;
+    wholeList.appendChild(choreList);
+  });
+};
+renderTasks();
+
+addBtn.addEventListener('click', (e) => {
+//   const store = (localStorage.getItem('store')) ? JSON.parse(localStorage.getItem('store')) : [];
```
Comment (samlorlah) [inline src/crud.js:27]: - [x] According to YAGNI rule, this should be removed please

**Review by samlorlah — CHANGES_REQUESTED** [review]

File: src/crud.js (hunk 3 above)
Comment (Recillah-Khamala) [reply src/crud.js:66]: done

File: src/crud.js (hunk 1 above)
Comment (Recillah-Khamala) [reply src/crud.js:37]: Done.

File: src/crud.js (hunk 4 above)
Comment (Recillah-Khamala) [reply src/crud.js:27]: Done

**Review by samlorlah — APPROVED** [review]
### STATUS: APPROVED!!!  🎉 🎉 🎉 🍾 🍾 🍾

Hi [@Rn486](https://github.com/Rn486),
Good job !!!

To Highlight! 👏 🟢
✔️ Good commit messages
✔️ Descriptive Readme file
✔️ Clean/Organized code

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 🎉 🎉 🍾 🍾 🍾

**Pull request merged**